File: ios/Playground/Sources/Theme/Shapes.swift
import SwiftUI

// MARK: - Playground shapes

/// Chunky hand-drawn corner scale, mirroring the Android `AutoMobileShapes`. The
/// token-layer foundation of the crayon look; true irregular "wobble" borders are
/// layered on in the component phase. Values are corner radii in points.
struct PlaygroundShapes {
    let extraSmall: CGFloat = 6
    let small: CGFloat = 12
    let medium: CGFloat = 18
    let large: CGFloat = 26
    let extraLarge: CGFloat = 40

    // Component-specific radii
    let button: CGFloat = 14
    let card: CGFloat = 18
    let textField: CGFloat = 12
    let dialog: CGFloat = 26
    let bottomSheet: CGFloat = 26

    func rounded(_ radius: CGFloat) -> RoundedRectangle {
        RoundedRectangle(cornerRadius: radius, style: .continuous)
    }
}

File: android/playground/design/system/src/main/kotlin/dev/jasonpearson/automobile/design/system/theme/Shapes.kt
package dev.jasonpearson.automobile.design.system.theme

import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Shapes
import androidx.compose.ui.unit.dp

// AutoMobile Design System Shapes.
// Chunkier, softer corners than the Material default scale — the token-layer
// foundation of the hand-drawn crayon look. True irregular "wobble" borders are
// layered on top in the design-system components (follow-up item). See
// docs/design-docs/playground-design-system.md.
val AutoMobileShapes =
  Shapes(
    extraSmall = RoundedCornerShape(6.dp),
    small = RoundedCornerShape(12.dp),
    medium = RoundedCornerShape(18.dp),
    large = RoundedCornerShape(26.dp),
    extraLarge = RoundedCornerShape(40.dp),
  )

// Additional custom shapes for specific use cases
object AutoMobileCustomShapes {
  val button = RoundedCornerShape(14.dp)
  val card = RoundedCornerShape(18.dp)
  val bottomSheet = RoundedCornerShape(topStart = 26.dp, topEnd = 26.dp)
  val dialog = RoundedCornerShape(26.dp)
  val textField = RoundedCornerShape(12.dp)
}
